integration/experimental/v3_zdt_push_command_test.go

```diff
@@ -554,6 +555,37 @@ var _ = Describe("v3-zdt-push command", func() {
 			})
 		})
 
+		Context("when the -s flag is set", func() {
+			var session *Session
+			BeforeEach(func() {
+				helpers.WithHelloWorldApp(func(appDir string) {
+					session = helpers.CustomCF(helpers.CFEnv{WorkingDirectory: appDir}, "v3-zdt-push", appName, "-s", "cflinuxfs2")
+					Eventually(session).Should(Exit(0))
+				})
+			})
+
+			It("uses the specified stack", func() {
+				Eventually(session).Should(Say("name:\\s+%s", appName))
+				Eventually(session).Should(Say("requested state:\\s+started"))
+				Eventually(session).Should(Say("routes:\\s+%s\\.%s", appName, domainName))
+				Eventually(session).Should(Say("stack:\\s+cflinuxfs2"))
```

Comment:
This is the default stack, so this example isn't really that great. We weren't sure if you regularly run these integration tests against multi-stack environments, though.

Additionally, if there's additional tests that you'd like to see please let us know. We were quite sure of where the boundaries between integration/unit testing lie within this codebase and would appreciate some guidance.

Thanks!